### DISCUSSION TITLE
Not getting badges even after meeting the conditions

### DISCUSSION THREAD

**Discussion by ARMAAN199** [community/community#62283] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I've met the conditions for "Pair Extraordinaire" and I've also met conditions for upgrading my "Pull Shark" badge but they don't seem to be updating. Is this a bug?

**Comment by nirav-gajera** [comment]
contact at github support by creating ticket they'll help you : https://support.github.com/

**Comment by sandroamiridze** [comment]
Caching: Sometimes, badge updates can be delayed due to caching mechanisms. GitHub might have caching systems in place to reduce server load and improve performance, which can cause delays in badge updates.

    Time Delay: Badge updates might not be instantaneous. The system may take some time to process and verify the conditions for earning a badge.

    GitHub Server Issues: Temporary server issues or outages on GitHub's end could also cause delays in badge updates.

    Badge Criteria: Double-check that you have met all the specific criteria for earning the "Pair Extraordinaire" and "Pull Shark" badges. Sometimes, there could be additional requirements that you may not have fulfilled.

    Bug or Glitch: In some cases, there might be a bug or glitch in GitHub's badge awarding system that is preventing the badges from updating correctly.

If you believe you have genuinely met all the conditions for the badges and they are still not updating after waiting for a reasonable amount of time, you can try the following steps:

    Log out and Log In: Sometimes, logging out of your GitHub account and then logging back in can trigger a refresh and update your badges.

    Clear Cache and Cookies: Clear your browser's cache and cookies, as old data may sometimes interfere with badge updates.

    Contact GitHub Support: If the issue persists, you can reach out to GitHub support to inquire about the badge update delay or potential issues with the badge awarding system.

**Reply by systemadmin-lab** [reply]
It is frustrating when your hard-earned achievements don't reflect on your profile immediately. While it might feel like a bug, GitHub's achievement system often has a "lag" or specific hidden requirements that prevent instant updates.

Here is a breakdown of why your Pair Extraordinaire and Pull Shark badges might not be updating and how to fix it.

1. The "Wait and See" Rule (24–48 Hours)
GitHub does not update achievement badges in real-time. The system typically processes these updates in batches.

The Delay: It can take anywhere from 24 to 48 hours after a qualifying event (like a PR merge) for the badge to recalculate.

Recommendation: If you just hit the milestone today, check back in two days.

2. Check Your "Private Contributions" Setting
By default, GitHub achievements often only account for public activity. If the PRs or co-authored commits you made were in private repositories, they won't count unless you enable this:

Go to your Profile Settings.

Look for "Contributions & Activity".

Ensure "Show private contributions on my profile" is checked.

3. Verify the Milestone Requirements
Sometimes we "feel" like we've hit the goal, but the math is specific.

Pull Shark: Only merged Pull Requests count. If a PR was "Closed" without being merged, it is ignored.

Tiers: 2 (Base), 16 (Bronze), 128 (Silver), 1024 (Gold).

Pair Extraordinaire: This requires you to be a co-author. In the commit message, there must be a specific trailer: Co-authored-by: name <email>. If the formatting was slightly off, GitHub’s parser might have missed it.

Tiers: 1 (Base), 10 (Bronze), 24 (Silver), 48 (Gold).

4. The "Hide and Unhide" Trick (Common Fix)
There is a known UI glitch where the badge level is updated in the backend but doesn't "refresh" on your public profile.

Go to your GitHub Profile.

Click on the Achievement itself.

Click the "Hiding" toggle (top right of the badge popup).

Wait a few seconds, then unhide it. This often forces the UI to pull the latest metadata from the server.

5. Check Your Email Configuration
For Pair Extraordinaire, your local Git email must match the email on your GitHub account.

Run git config user.email in your terminal.

Verify this email is added and verified in your GitHub [Email Settings](https://github.com/settings/emails). If GitHub doesn't recognize the email in the commit, it won't credit you for the co-authorship.

**Comment by YasinDehfuli** [comment]
If you need help to get new GitHub achievements, You can use this repository to get new achievements step by step:

https://github.com/4xmen/Get-Github-Achievements

**Comment by systemadmin-lab** [comment]
It is frustrating when your hard-earned achievements don't reflect on your profile immediately. While it might feel like a bug, GitHub's achievement system often has a "lag" or specific hidden requirements that prevent instant updates.

Here is a breakdown of why your Pair Extraordinaire and Pull Shark badges might not be updating and how to fix it.

1. The "Wait and See" Rule (24–48 Hours)
GitHub does not update achievement badges in real-time. The system typically processes these updates in batches.

The Delay: It can take anywhere from 24 to 48 hours after a qualifying event (like a PR merge) for the badge to recalculate.

Recommendation: If you just hit the milestone today, check back in two days.

2. Check Your "Private Contributions" Setting
By default, GitHub achievements often only account for public activity. If the PRs or co-authored commits you made were in private repositories, they won't count unless you enable this:

Go to your Profile Settings.

Look for "Contributions & Activity".

Ensure "Show private contributions on my profile" is checked.

3. Verify the Milestone Requirements
Sometimes we "feel" like we've hit the goal, but the math is specific.

Pull Shark: Only merged Pull Requests count. If a PR was "Closed" without being merged, it is ignored.

Tiers: 2 (Base), 16 (Bronze), 128 (Silver), 1024 (Gold).

Pair Extraordinaire: This requires you to be a co-author. In the commit message, there must be a specific trailer: Co-authored-by: name <email>. If the formatting was slightly off, GitHub’s parser might have missed it.

Tiers: 1 (Base), 10 (Bronze), 24 (Silver), 48 (Gold).

4. The "Hide and Unhide" Trick (Common Fix)
There is a known UI glitch where the badge level is updated in the backend but doesn't "refresh" on your public profile.

Go to your GitHub Profile.

Click on the Achievement itself.

Click the "Hiding" toggle (top right of the badge popup).

Wait a few seconds, then unhide it. This often forces the UI to pull the latest metadata from the server.

5. Check Your Email Configuration
For Pair Extraordinaire, your local Git email must match the email on your GitHub account.

Run git config user.email in your terminal.

Verify this email is added and verified in your GitHub [Email Settings](https://github.com/settings/emails). If GitHub doesn't recognize the email in the commit, it won't credit you for the co-authorship.

**Comment by ADITYAchhipa** [comment]
It does sound like a bug. Badge updates are usually automated, but sometimes they don’t trigger immediately. I recommend:

Wait a few hours and refresh your profile.

If it still doesn’t update, report it via the official support/contact form so a team member can manually verify and fix it.
